exchanges/graphcache/src/operations/query.ts

```diff
@@ -373,7 +392,7 @@ const readSelection = (
     ctx.__internal.path.push(fieldAlias);
     // We temporarily store the data field in here, but undefined
     // means that the value is missing from the cache
-    let dataFieldValue: void | DataField;
+    let dataFieldValue: void | DataField = undefined;
```

Comment:
had to do this because TS kept thinking it wasn't being assigned

Reply:
```suggestion
    let dataFieldValue: undefined | DataField;
```

This usually does the trick. Not sure when TS changed this, could've been a long time ago or always been there, but I know that `undefined` allows unassigned undefined-defaulted variables, `void` does not